src/shared/filesystemUtilities.ts

```diff
@@ -71,16 +66,38 @@ export async function findFileInParentPaths(searchFolder: string, fileToFind: st
     return findFileInParentPaths(parentPath, fileToFind)
 }
 
-const mkdtemp = promisify(fs.mkdtemp)
+let tmpDirRoot: string // lazily initialized
+export async function getTemporaryToolkitFolderRoot() {
+    if (!tmpDirRoot /* never initialized */ || !await fileExists(tmpDirRoot) /* has been deleted */) {
+        const _tempDirPath = path.join(
+            // https://github.com/aws/aws-toolkit-vscode/issues/240
+            os.type() === 'Darwin' ? '/tmp' : os.tmpdir(),
+            'aws-toolkit-vscode'
+        )
+        tmpDirRoot = await new Promise<string>((resolve, reject) => {
+            tmp.dir(
+                {
+                     dir: _tempDirPath,
+                 },
+                (err: Error, _path: string) => {
+                    if (err) {
+                        reject(err)
+                    }
+                    resolve(_path)
+                }
+            )
+        })
+    }
+
+    return tmpDirRoot
+}
+
 export const  makeTemporaryToolkitFolder = async (...relativePathParts: string[]) => {
     const _relativePathParts = relativePathParts || ['vsctk']
-    const tmpPath = path.join(tempDirPath, ..._relativePathParts)
-    const tmpPathParent = path.dirname(tmpPath)
-    // fs.makeTemporaryToolkitFolder fails on OSX if prefix contains path separator
-    // so we must create intermediate dirs if needed
-    if (!await fileExists(tmpPathParent)) {
-        await mkdir(tmpPathParent, { recursive: true })
+    const tmpPath = path.join(await getTemporaryToolkitFolderRoot(), ..._relativePathParts)
+    if (!await fileExists(tmpPath)) {
+        await mkdir(tmpPath)
```

Comment:
Looks like there's a fair amount of issues in the tests where mkdir is failing due to paths not existing. Consider making this recursive? This should clear up everything relating to the logger (most of the "before all" and "after all" failures).

Reply:
Oh thanks! Interesting that it worked on OSX but not Windows. Fixed.
✔️